lib/plugins/aws/package/compile/events/eventBridge/index.js

```diff
@@ -0,0 +1,155 @@
+
+            if (InputTransformer) {
+              const config = Object.assign({}, InputTransformer);
+              InputTransformer = {};
+              if (!config.inputTemplate) {
+                throw new this.serverless.classes.Error(
+                  'The inputTemplate key is required when specifying an ' +
+                    'inputTransformer for an eventBridge event'
+                );
+              }
+              InputTransformer.InputTemplate = config.inputTemplate;
+              if (config.inputPathsMap) {
+                InputTransformer.InputPathsMap = config.inputPathsMap;
+              }
+            }
+
+            const customEventBridge = {
+              [customEventBridgeResourceLogicalId]: {
+                Type: 'Custom::EventBridge',
+                Version: 1.0,
+                DependsOn: [eventFunctionLogicalId, customResourceFunctionLogicalId],
+                Properties: {
+                  ServiceToken: {
+                    'Fn::GetAtt': [customResourceFunctionLogicalId, 'Arn'],
+                  },
+                  FunctionName,
+                  EventBridgeConfig: {
+                    RuleName,
+                    EventBus,
+                    Schedule,
+                    Pattern,
+                    Input,
+                    InputPath,
+                    InputTransformer,
+                  },
+                },
+              },
+            };
+
+            _.merge(compiledCloudFormationTemplate.Resources, customEventBridge);
+
+            let eventBusResource;
+            let ruleResource = `arn:aws:events:*:*:rule/${RuleName}`;
+            if (EventBus) {
+              let eventBusName = EventBus;
+              if (EventBus.startsWith('arn')) {
+                eventBusName = EventBus.split('/').pop();
+              }
+              eventBusResource = `arn:aws:events:*:*:event-bus/${eventBusName}`;
+              ruleResource = `arn:aws:events:*:*:rule/${eventBusName}/${RuleName}`;
+              iamRoleStatements.push({
+                Effect: 'Allow',
+                Resource: eventBusResource,
+                Action: ['events:CreateEventBus', 'events:DeleteEventBus'],
+              });
+            }
+
+            iamRoleStatements.push({
+              Effect: 'Allow',
+              Resource: ruleResource,
+              Action: [
+                'events:PutRule',
+                'events:RemoveTargets',
+                'events:PutTargets',
+                'events:DeleteRule',
+              ],
+            });
+          } else {
+            const errorMessage = [
+              `Event Bridge event of function "${functionName}" is not an object`,
+              ' Please check the docs for more info.',
+            ].join('');
+            throw new this.serverless.classes.Error(errorMessage);
+          }
+        });
+      }
+
+      iamRoleStatements.push({
+        Effect: 'Allow',
+        Resource: `arn:aws:lambda:*:*:function:${FunctionName}`,
```

Comment:
shouldn't we set the region & account id?

Reply:
Oh yeah, I tried to do that but getting access to the account id is quite tedious. I can look into that tomorrow.

Reply:
```suggestion
        Resource: {
          'Fn::Join': [
            ':',
            [
              'arn:aws:lambda',
              { Ref: 'AWS::Region' },
              { Ref: 'AWS::AccountId' },
              'function',
              FunctionName,
            ]
          }
```

Reply:
![image](https://user-images.githubusercontent.com/1606004/61823843-e1cc5600-ae5c-11e9-97ab-20981ba9ffb1.png)

Thanks! 👌

That happens when you spend too much time in the programmatic realm.

Reply:
I didn't remember it, I just knew I had projects that did similar and where to look: https://github.com/dschep/bikehero/blob/master/serverless.yml#L17-L22

Reply:
> knowing where to find an answer can be just as valuable as knowing the answer 🙂

👌 💯